### PR TITLE
build: make root.tsx optional

### PR DESCRIPTION
**Pull request by shairez**
# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests

# Description

Make build `root.tsx` for client mode building to make it easier for testing tools to run Qwik libraries.

### CODE REVIEW

**Comment by mhevery** [comment]
@shairez can you better explain the motivation?

**Comment by shairez** [comment]
Yes, it's something me and @manucorporat figured out when trying to build the cypress plugin.

I ran into some limitations when trying to fix this, but I need to complete this work when I'll have time

File: packages/qwik/src/optimizer/src/plugins/plugin.ts [hunk 1]
```diff
@@ -758,6 +759,16 @@ export const isDev = ${JSON.stringify(isDev)};
...
   }
 
+  async function isPathExist(path: string) {
+    const sys = getSys();
+    if (sys.env === 'node') {
+      const fs: typeof import('fs') = await sys.dynamicImport('node:fs');
+
+      return fs.existsSync(path);
+    }
+    return false;
```
Comment (wmertens) [inline packages/qwik/src/optimizer/src/plugins/plugin.ts:769]: maybe better to assume they exist, so that it's like previous behavior?

**Comment by PatrickJS** [comment]
oh I actually ran into this when I was working on Qwik Start. It should be optional

**Comment by PatrickJS** [comment]
can we redo this PR

**Comment by wmertens** [comment]
we're going to do this as part of the vite environment support

**Pull request closed without merging**